src/platform/datapath_raw_socket.c

```diff
@@ -54,7 +55,8 @@ CxPlatGetSocket(
     Entry = CxPlatHashtableLookup(&Pool->Sockets, LocalAddress->Ipv4.sin_port, &Context);
     while (Entry != NULL) {
         CXPLAT_SOCKET* Temp = CONTAINING_RECORD(Entry, CXPLAT_SOCKET, Entry);
-        if (QuicAddrCompareIp(&Temp->LocalAddress, LocalAddress)) {
+        if (QuicAddrCompareIp(&Temp->LocalAddress, LocalAddress) &&
+            (!Temp->Connected || QuicAddrCompareIp(&Temp->RemoteAddress, RemoteAddress))) {
```

Comment:
```suggestion
            (!Temp->Connected || QuicAddrCompare(&Temp->RemoteAddress, RemoteAddress))) {
```
As Matt mentioned, this should compare both IP and port. The other one already has compared port because of the call to `CxPlatHashtableLookup` takes the port.